### PR TITLE
HOTT-846: Add Rules of Origin API client

### PR DESCRIPTION
**Pull request by jebw**
### Jira link

[HOTT-846](https://transformuk.atlassian.net/browse/HOTT-846)

### What?

I have added/removed/altered:

- [x] Added an API client for the `/rules_of_origin_schemes` endpoint
- [x] Disabled feature flag in `Rails.env.development?` - the logic being other backend branches may not yet have latest api changes - this avoids `master` frontend having a hard dependency on tip of `master` backend. Developers can enable rules of origin by setting feature flag locally.

### Why?

I am doing this because:

- We should be querying the backend rather then showing mocked data from the frontend

### Notes

This supercedes #195

### CODE REVIEW

File: app/models/goods_nomenclature.rb [hunk 1]
```diff
@@ -51,4 +51,14 @@ def id
   def code
     goods_nomenclature_item_id
   end
+
+  def rules_of_origin(*args, **kwargs)
+    return nil unless try(:declarable?)
```
Comment (willfish) [inline app/models/goods_nomenclature.rb]: Don't we always have this answer?

File: spec/models/goods_nomenclature_spec.rb [hunk 2]
```diff
@@ -92,4 +92,72 @@
       end
     end
   end
+
+  describe '#rules_of_origin' do
+    before do
+      # Needed because goods nomenclature isn't declarable but #rules_of_origin
```
Comment (willfish) [inline spec/models/goods_nomenclature_spec.rb:98]: I think we can reflect that section, chapters, etc, all have declarable and just treat them as ducks/trust the data

**Review by willfish — APPROVED** [review]
Love it

File: app/models/goods_nomenclature.rb (hunk 1 above)
Comment (jebw) [reply app/models/goods_nomenclature.rb]: Not on `Chapter` which also inherits from `GoodsNomenclature`

File: spec/models/goods_nomenclature_spec.rb (hunk 2 above)
Comment (jebw) [reply spec/models/goods_nomenclature_spec.rb:98]: Ok, will change `Chapter` to always return false for `#declarable?`

Comment (jebw) [reply spec/models/goods_nomenclature_spec.rb:98]: This override will still be required though because we want to bypass the short-circuit

**Review by willfish — APPROVED** [review]

**Review by willfish — APPROVED** [review]

**Pull request merged**